packaging/build-debian-copyright

```diff
@@ -0,0 +1,216 @@
+    parts = [p.strip() for p in license_text.split(' and ')]
+
+    normalized_parts: list[str] = []
+    for part in parts:
+        normalized = normalize_part(part)
+        if normalized.lower() not in license_ids:
+            sys.exit(f"ERROR: License '{normalized}' is not defined in {TEMPLATE_FILE}\n"
+                     f"Original license text: '{license_text}'\n"
+                     "Please add a License: paragraph for this license to the template.")
+        normalized_parts.append(normalized)
+
+    return ' and '.join(normalized_parts)
+
+
+def get_package_info(package_path: str, files_data: dict[str, str], license_ids: set[str]) -> tuple[str, set[str]]:
+    """Extract license and copyright info from package file data"""
+    copyrights: set[str] = set()
+
+    # Read package.json for license and author
+    package_json_key = f"{package_path}/package.json"
+    if package_json_key not in files_data:
+        sys.exit(f"ERROR: No package.json found for {package_path}")
+
+    pkg_data = json.loads(files_data[package_json_key])
+    license_text = normalize_spdx_license(pkg_data['license'], license_ids)
+
+    # Get author from package.json
+    if author := pkg_data.get('author'):
+        if isinstance(author, dict):
+            author_dict = cast("dict[str, str]", author)
+            author_name = str(author_dict.get('name', ''))
+        else:
+            # Parse "Name <email>" format
+            author_name = re.sub(r'\s*<[^>]+>\s*', '', str(author)).strip()
+        copyrights.add(author_name)
+
+    # Look for LICENSE files (various names)
+    for license_file in ['LICENSE', 'LICENSE.md', 'LICENSE.txt', 'COPYING', 'COPYING.txt']:
+        license_key = f"{package_path}/{license_file}"
+        if license_key in files_data:
+            copyrights.update(find_copyright_in_license_text(files_data[license_key]))
+
+    return license_text, copyrights
+
+
+def extract_files_from_tarball(tarball_path: Path) -> tuple[dict[str, str], dict[str, set[str]]]:
+    """Extract package.json and LICENSE files from node_modules tarball.
+
+    Returns:
+        files_data: {"pkg/file" -> content}
+        toplevel_pkgs: {toplevel -> {pkg_paths}}
+    """
+    files_data: dict[str, str] = {}
+    toplevel_pkgs: dict[str, set[str]] = {}
+
+    with tarfile.open(tarball_path) as tar:
+        for member in tar.getmembers():
+            if not member.isfile() or not member.name.startswith('node_modules/'):
+                continue
+
+            # node_modules/pkg/file or node_modules/@scope/pkg/file
+            parts = member.name.split('/')
+            if parts[1].startswith('@'):
+                if len(parts) != 4:  # must be node_modules/@scope/pkg/file
+                    continue
+                pkg_path, filename = '/'.join(parts[1:3]), parts[3]
+            else:
+                if len(parts) != 3:  # must be node_modules/pkg/file
+                    continue
+                pkg_path, filename = parts[1], parts[2]
```

Comment:
I think this could be radically simplified now: all we care about, really, are 3 things:

 - `node_modules` (drop this)
 - the very first component that follows that ("toplevel package")
 - (ignoring everything else between)
 - the basename (so we can check our list)

This way we'd also find license files located at

`node_modules/projname/src/COPYING`

just as well as

`node_modules/@group/prog/COPYING`

and in all cases they're gonna land int the correct `toplevel_pkgs`.